postgres/src/main/java/org/jdbi/v3/pg/ArrayColumnMapper.java

```diff
@@ -34,17 +34,23 @@ public ArrayColumnMapper(Class<?> componentType, StatementContext ctx) {
     }
 
     @Override
-    public Object[] map(ResultSet r, int columnNumber, StatementContext ctx) throws SQLException {
+    public Object map(ResultSet r, int columnNumber, StatementContext ctx) throws SQLException {
         return buildArray(r.getArray(columnNumber));
     }
 
-    private Object[] buildArray(Array array) throws SQLException {
+    private Object buildArray(Array array) throws SQLException {
         if (UNSUPPORTED_TYPES.contains(array.getBaseType())) {
             return buildFromResultSet(array);
         }
         try {
-            // more efficient, doesn't have to copy an unknown size array
-            return (Object[]) array.getArray();
+            // We can't cast Object[] to componentType, so we need to create a new array of the correct
+            // type and copy the source array content to it.
+            Object[] objectArray = (Object[]) array.getArray();
+            Object componentTypeArray = java.lang.reflect.Array.newInstance(componentType, objectArray.length);
+            for (int i = 0; i < objectArray.length; i++) {
```

Comment:
Could we skip this loop if the component type of the object array is already correct?